client/src/modules/templates/bhCardList.tmpl.html

```diff
@@ -0,0 +1,45 @@
+<div class="panel panel-default bh-card-list">
+  <div class="panel-heading">
+    <a href ng-click="$ctrl.toggleFilter()">
+      <span translate>TABLE.COLUMNS.FILTERING.LABEL</span>
+      <span class="caret" ng-class="{'rotate' : $ctrl.filterOptions.active}"></span>
+    </a>
+
+    <div class="pull-right">
+      <span class="label label-primary" style="margin-right: 5px;" ng-if="$ctrl.orderOptions.active">
+        <span translate>{{$ctrl.orderOptions.current.key}}</span>
+      </span>
+      <span uib-dropdown>
+        <a href uib-dropdown-toggle> <span translate>TABLE.COLUMNS.SORTING.LABEL</span> <span class="caret"></span></a>
+        <ul class="dropdown-menu-right textflow" uib-dropdown-menu>
+          <li ng-repeat="option in $ctrl.orderOptions.available">
+            <a href ng-click="$ctrl.setOrder(option)">
+              <span translate>{{ option.key }}</span>
+            </a>
+          </li>
+          <li role="seperator" class="divider"></li>
+          <li>
+            <a href ng-click="$ctrl.setOrder()">
+              <span class="fa fa-times" aria-hidden="true"></span>
+              <span translate> FORM.BUTTONS.CLEAR </span>
+            </a>
+          </li>
+        </ul>
+      </span>
+    </div>
+  </div>
+
+  <input
+    ng-model="$ctrl.filterOptions.value[$ctrl.currentFilterAttribute]"
+    ng-show="$ctrl.filterOptions.active"
+    class="form-control"
+    placeholder="{{'FORM.PLACEHOLDERS.FILTER_NAME' | translate }}"
```

Comment:
Just FYI - with any plain HTML element, you can use `angular-translate` attributes for a slight performance increase, instead of using the expensive pipe.  However, this does not work for custom components (read: `ui-select`), which is why you still see this pattern in those components.

To use the translate attributes, you can use the [translate-attr](https://angular-translate.github.io/docs/#/api/pascalprecht.translate.directive:translate-attr) directive to do it in one pass:
```html
<input
  translate-attr="{ placeholder : 'FORM.PLACEHOLDERS.FILTER_NAME' }">
```

Reply:
Good to know!